src/framework/execution/types/instance.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * A stored user configuration for executing the integration defined by
+ * associated the `integrationDefinitionId`.
+ */
+export interface IntegrationInstance {
+  /**
+   * Unique identifier for the activated integration instance.
+   */
+  id: string;
+
+  /**
+   * `accountId` identifies the tenant/account holder that activated the
+   * integration.
+   */
+  accountId: string;
+
+  /**
+   * A short friendly name for the integration instance that is provided by
+   * end-user.
+   */
+  name: string;
+
+  /**
+   * Optional description of the integration instance.
+   */
+  description?: string;
+
+  /**
+   * The `integrationDefinitionId` identifies the integration definition
+   * that the instance relates to.
+   */
+  integrationDefinitionId: string;
+
+  /**
+   * Each integration specifies the properties it requires a user provide when
+   * configuring an instance of the integration. It is up to the UI to validate
+   * input at configuration time, and the integration should also validate the
+   * configuration upon invocation and provide useful configuration error
+   * messages.
+   */
+  config: any;
+
+  /**
+   * Ternary marker for offsite flow state.
+   *
+   * Integrations supporting an offsite installation flow will set this value to
+   * `false` when the instance is created, indicating that the configuration is
+   * incomplete. The value will become `true` when the offsite flow has
+   * completed successfully.
+   *
+   * Integrations that do/will not support an offsite installation flow will
+   * leave this value `undefined`.
+   */
+  offsiteComplete?: boolean;
```

Comment:
I'm not totally sure, but I wonder if this should be in the types contained in the integration SDK. I believe this property is not seen by the integration itself, but is used by J1 to manage offsite flow. It's super convenient to store it on the instance, but isn't a concern outside of J1. What do you think of leaving it out so it isn't something integration developers have to care about?

@fomentia please correct me if I'm wrong, but I don't see any references to this in the GitHub integration.

Reply:
Ah, I probably should have checked the github integration before blindly adding this. I thought this was used for preventing an integration from doing anything when invoked, but I now see that this work is done by the scheduler. I'll remove it.